parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoParquetWriter.java

```diff
@@ -82,42 +84,70 @@ public ProtoParquetWriter(Path file, Class<? extends Message> protoMessage) thro
     this(file, protoMessage, CompressionCodecName.UNCOMPRESSED,
             DEFAULT_BLOCK_SIZE, DEFAULT_PAGE_SIZE);
   }
-  
-  public static <T> Builder<T> builder(Path file) {
-	    return new Builder<T>(file);
-	}
-
-	public static <T> Builder<T> builder(OutputFile file) {
-	    return new Builder<T>(file);
-	}
-	
-	private static <T extends MessageOrBuilder> WriteSupport<T> writeSupport(Class<? extends Message> protoMessage) {
-		return new ProtoWriteSupport<T>(protoMessage);
-	}
-	  
-	public static class Builder<T> extends ParquetWriter.Builder<T, Builder<T>> {
-		  
-		Class<? extends Message> protoMessage = null;
-
-		private Builder(Path file) {
-			super(file);
-		}
-
-		private Builder(OutputFile file) {
-		    super(file);
-		}
-
-		protected Builder<T> self() {
-		    return this;
-		}
-		
-		public Builder<T> withMessage(Class<? extends Message> protoMessage){
-			this.protoMessage = protoMessage;
-			return this;
-		}
-
-		protected WriteSupport<T> getWriteSupport(Configuration conf) {
-		    return (WriteSupport<T>) ProtoParquetWriter.writeSupport(protoMessage);
-		}    
-	}
+
+  public static <T extends MessageOrBuilder> Builder<T> builder(Path file) {
```

Comment:
@gszadovszky If your concern is that a client could have been successfully using a type parameter that didn't extend MessageOrBuilder I am almost certain that's not the case.  MessageOrBuilder is one of the highest protobuf abstractions and I can't imagine a running use of this that was outside that interface.

Reply:
@dossett, I accept that but we have a maven plugin (japicmp) that checks compatibility. I do not want to add exclusions for this. Let's see if the build passes (after a rebase). If yes, I am fine with that if not, we will need to implement this change in another way.

Reply:
That makes sense.  From some experimenting it seems the `extends MessageOrBuilder` can be dropped with no harm, it would just be a nice cue to someone reading the code.